tls/s2n_psk.c

```diff
@@ -278,55 +306,70 @@ S2N_RESULT s2n_finish_psk_extension(struct s2n_connection *conn)
     return S2N_RESULT_OK;
 }
 
-static S2N_RESULT s2n_psk_set_hmac(struct s2n_psk *psk, s2n_psk_hmac psk_hmac_alg)
+int s2n_psk_set_hmac(struct s2n_psk *psk, s2n_psk_hmac hmac)
 {
-    switch(psk_hmac_alg) {
+    notnull_check(psk);
+    switch(hmac) {
         case S2N_PSK_HMAC_SHA224:     psk->hmac_alg = S2N_HMAC_SHA224; break;
         case S2N_PSK_HMAC_SHA256:     psk->hmac_alg = S2N_HMAC_SHA256; break;
         case S2N_PSK_HMAC_SHA384:     psk->hmac_alg = S2N_HMAC_SHA384; break;
         default:
-            BAIL(S2N_ERR_HMAC_INVALID_ALGORITHM);
+            S2N_ERROR(S2N_ERR_HMAC_INVALID_ALGORITHM);
     }
-    return S2N_RESULT_OK;
+    return S2N_SUCCESS;
 }
 
-int s2n_connection_set_external_psks(struct s2n_connection *conn, struct s2n_external_psk *psk_vec, size_t psk_vec_length)
+int s2n_connection_set_external_psks(struct s2n_connection *conn, struct s2n_psk **psk_list, uint16_t psk_list_len)
```

Comment:
What is the rational behind changing psk_vec_length from a size_t to uint16_t?

Reply:
The thinking was that we can't send more than uint16_t psks over the wire, but since the s2n_array struct uses a uint32_t for its length, that's probably a more accurate limit.